### PR TITLE
Add documentation for the new login flow

### LINKED ISSUE #620: "New" weblogin flow needs documentation
The new login flow should be documented so that third party devs can easily use this without reading android/ios/client/server code.

Ref: https://github.com/nextcloud/android/issues/1145#issuecomment-348210849

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #620

@MariusBluem please have a look and make it look nice :)

### CODE REVIEW

**Comment by marinofaggiana** [comment]
Why the login flow do not return the protocol (http or https) on server parameter ?

File: developer_manual/client_apis/LoginFlow/index.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+To have a good user experince please to considet he following things:
+* set the proper :code:`ACCEPT_LANGUAGE` header
+* set a proper :code:`USER_AGENT` header
+
+
+Login in the user
+-----------------
+
+The user will now see a webpage telling them they will grant access to :code:`USER_AGENT`. When they follow the setps
```
Comment (oparoz) [inline developer_manual/client_apis/LoginFlow/index.rst]: steps

**Comment by rullzer** [comment]
@marinofaggiana ah good point. 12 servers return no protocol. 13 does. Let me update that.

**Comment by mario** [comment]
@rullzer just to mention it here too. 12 also doesn't include subfolders. say "test.com/bla" would return just "test.com"  in 12.

**Comment by rullzer** [comment]
@mario thnx added a note

File: developer_manual/client_apis/LoginFlow/index.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	nc://login/server:<server>&user:<username>&password:<password>
+
+* server: The address of the server to connect to. The server may specify a protocol (http or https). If no protocol is specified the client will assume https.
+* username: The username that the client must use to login
+* password: The password that the client must use to login and store securely
+
+This information will be used by the client to create a new account.
+After this the webview is destroyed including all the state the webview holds.
+
+.. note:: On nextcloud 12 the returned server is just the server address without any possible subfolder. This is corrected in Nextcloud 13.
```
Comment (splitt3r) [inline developer_manual/client_apis/LoginFlow/index.rst]: `nextcloud 12` and correct `Nextcloud 13` looks strange 😁 both ucfirst would look much better.

**Comment by rullzer** [comment]
@MariusBluem can you finish (if there is something to finish) this and then lets merge it :)

**Comment by MariusBluem** [comment]
Fine with me - Screenshots of the login flow will be added by a separate PR.

**Comment by MorrisJobke** [comment]
[stable12 4e298523] Add documentation for the new login flow
[stable12 c12c302a] Typos
[stable12 2e0646fe] Explain parameters
[stable12 cf7807f7] Note about subfolder
[stable12 e732cecd] Fix typo

**Pull request merged**